crypto/core_namemap.c

```diff
@@ -170,7 +172,8 @@ int ossl_namemap_name2num_n(const OSSL_NAMEMAP *namemap,
     if (namemap == NULL)
         return 0;
 
-    CRYPTO_THREAD_read_lock(namemap->lock);
+    if (!CRYPTO_THREAD_read_lock(namemap->lock))
+        return 0;
```

Comment:
0 does not seem like a good way to return an error.

Reply:
> 0 does not seem like a good way to return an error.

but it is how the function is written.  not my doing :)